src/main/java/uk/gov/companieshouse/api/testdata/service/impl/CompanyPscsServiceImpl.java

```diff
@@ -19,165 +19,293 @@
-        return existingPscs.isPresent();
+    private Instant getCreatedUpdatedAt(CompanySpec spec) {
+        if (StringUtils.hasText(spec.getAccountsDueStatus())) {
+            LocalDate dueDateNow = randomService.generateAccountsDueDateByStatus(
+                    spec.getAccountsDueStatus());
+            return dueDateNow.atStartOfDay(ZoneId.of("UTC")).toInstant();
+        }
+        return LocalDate.now().atStartOfDay(ZoneId.of("UTC")).toInstant();
     }
 
-    private CompanyPscs differentiatePsc(CompanyPscs companyPsc) {
+    private void buildIndividualBeneficialOwner(CompanyPscs beneficialOwner) {
+        beneficialOwner.setKind(PscType.INDIVIDUAL_BENEFICIAL_OWNER.getKind());
+        beneficialOwner.setCountryOfResidence(addressService.getCountryOfResidence(
+                Jurisdiction.WALES));
+        beneficialOwner.setNationality(NATIONALITY);
+        beneficialOwner.setDateOfBirth(new DateOfBirth(20, 9, 1975));
 
-        String pscType;
-        String linkType;
-        Optional<List<CompanyPscs>> existingPscs = repository.findByCompanyNumber(companyPsc.getCompanyNumber());
+        beneficialOwner.setNameTitle("Dr.");
+        beneficialOwner.setNameForename("Dray");
+        beneficialOwner.setNameSurname("Day");
+        beneficialOwner.setName("Dr. Dray Day");
 
-        switch (existingPscs.orElse(new ArrayList<>()).size()) {
-            case 0:
-                pscType = "individual-person-with-significant-control";
-                linkType = "individual";
-                buildIndividualPsc(companyPsc, pscType, linkType);
-                break;
-            case 1:
-                pscType = "legal-person-person-with-significant-control";
-                linkType = "legal-person";
-                buildWithIdentificationPsc(companyPsc, pscType, linkType);
-                break;
-            default:
-                pscType = "corporate-entity-person-with-significant-control";
-                linkType = "corporate-entity";
-                buildWithIdentificationPsc(companyPsc, pscType, linkType);
-        }
+        beneficialOwner.setSanctioned(false);
+        beneficialOwner.setIsSanctioned(false);
 
-        return companyPsc;
+        var links = new Links();
+        links.setSelf(URL_PREFIX + beneficialOwner.getCompanyNumber()
+                + PSC_SUFFIX + PscType.INDIVIDUAL_BENEFICIAL_OWNER.getLinkType()
+                + "/" + beneficialOwner.getId());
+
+        beneficialOwner.setUsualResidentialAddress(addressService.getAddress(Jurisdiction.NON_EU));
+        beneficialOwner.setResidentialAddressSameAsServiceAddress(false);
     }
 
-    private CompanyPscs buildIndividualPsc(CompanyPscs companyPsc, String pscType, String linkType) {
+    private void buildCorporateBeneficialOwner(CompanyPscs beneficialOwner) {
+        beneficialOwner.setKind(PscType.CORPORATE_BENEFICIAL_OWNER.getKind());
+        beneficialOwner.setName("James Bond");
```

Comment:
I'd avoid using James Bond here and perhaps use `"FirstName SecondName"`. Its clearer should you have to manipulate the String at a later date and doesn't use a name covered by Copyright.